### DISCUSSION TITLE
is_isomorphic() function

### DISCUSSION THREAD

**Discussion by Harry4FunResearch** [pyg-team/pytorch_geometric#4530] [Q&A] [answered]
Is there a function similar to is_isomorphic() for comparing two graphs?

**Comment by rusty1s** [comment, marked as answer]
We do not support any `is_isomorphic()` functionality since PyG is a deep learning/GNN library rather than a general graph analytics library. IMO, you are better off to convert PyG graphs to a general graph analytics library (e.g., `networkx`) and perform the checks their. If you are only interested in approximating isomorphism, you might be able to get something to work via the `torch_geometric.nn.conv.WLConv` to model the Weisfeiler-Lehman graph isomorphism test by comparing the histogram of node colors of two graphs (see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/wl_kernel.py) for an example).

**Reply by Harry4FunResearch** [reply]
Thank you so much Matthias for the informative answer! I wanted to approximate the graph isomorphism issue in a batch, so WLConv may work for me. I think I can leverage the inner product of the histogram vectors generated.

**Reply by Harry4FunResearch** [reply]
I tried to import WLConv by using "from torch_geometric.nn.conv import WLConv", but got the error of "ImportError: cannot import name 'WLConv'". 
Does it mean my Geometric version is too old?

**Reply by rusty1s** [reply]
Yes. Which PyG version are you using?
```python
import torch_geometric
print(torch_geometric.__version__)
```

**Reply by Harry4FunResearch** [reply]
Thanks Matthias. It was 1.6.3. I just updated to the latest version, and it works perfectly. BTW, for two graphs to be isomorphic, their histograms resulting from this function need to be identical?

**Reply by rusty1s** [reply]
Yes, that is correct.

**Reply by Harry4FunResearch** [reply]
Really appreciate your help Matthias! BTW, is there a version of WLConv that takes into account the edge weights?

**Reply by rusty1s** [reply]
`WLConv` is a discrete operator, so it cannot take edge weights into account :(

**Reply by Harry4FunResearch** [reply]
yes. it makes sense.  Thanks Matthias.
